### ISSUE TITLE
Add git so that dart pub get works for pubspecs with git dependencies

### ISSUE THREAD

**Issue by cpswan** [dart-lang/dart-docker#26]
At present git isn't part of the base image or included in the Dockerfile template, so when a Dockerfile using `FROM dart` tries to run `dart pub get` the result is a sequence of:

```
Git command is not "git": Pub failed to run subprocess `git`: ProcessException: No such file or directory
  Command: git --version
Git command is not "git.cmd": Pub failed to run subprocess `git.cmd`: ProcessException: No such file or directory
  Command: git.cmd --version
```

This can be fixed by adding git to the `apt-get` section of the Dockerfile e.g.

```dockerfile
RUN set -eux; \
    apt-get update; \
    apt-get install -y --no-install-recommends \
        ca-certificates \
        curl \
        openssh-client \
        unzip \
        dnsutils \
        git \
    ; \
```

The overall package diff (comparing `apt list --installed` between the existing dart image and one with git added) ends up being:

```
22a23,24
> git-man/now 1:2.20.1-2+deb10u3 all [installed,local]
> git/now 1:2.20.1-2+deb10u3 amd64 [installed,local]
41a44
> libcurl3-gnutls/now 7.64.0-4+deb10u2 amd64 [installed,local]
46a50,51
> liberror-perl/now 0.17027-2 all [installed,local]
> libexpat1/now 2.2.6-2+deb10u1 amd64 [installed,local]
52a58,59
> libgdbm-compat4/now 1.18.1-4 amd64 [installed,local]
> libgdbm6/now 1.18.1-4 amd64 [installed,local]
84a92
> libpcre2-8-0/now 10.32-5 amd64 [installed,local]
85a94
> libperl5.28/now 5.28.1-6+deb10u1 amd64 [installed,local]
117a127,128
> perl-modules-5.28/now 5.28.1-6+deb10u1 all [installed,local]
> perl/now 5.28.1-6+deb10u1 amd64 [installed,local]
```

resulting in an extra 76MB used by the raw image, which becomes 34MB when compressed on Docker Hub (for x64 arch).

Obviously people can workaround this by putting `apt-get install -y git` into their own Dockerfiles using `FROM dart`, and even create their own build images with git added. But as `dart pub get` is such a common thing to run, the various trade offs of image size, build times etc. would favour having a key ecosystem dependency like git in place.

**Comment by subfuzion** [comment]
@cpswan I'm unable to replicate this. I can enter a dart container (`docker run -it --rm -v $PWD:/app -w /app --entrypoint bash dart`) from a directory with a project and run `dart pub get`. Also, all the Dart Functions Framework [examples](https://github.com/GoogleCloudPlatform/functions-framework-dart/tree/main/examples) use `dart:stable` as the base image and these all continue to work (and pass our CI tests).

Could you provide more detail about what's happening for you (perhaps with a link to a  gist or repo)?

**Comment by cpswan** [comment]
@subfuzion - Looking deeper into the [pubspec.yaml](https://github.com/atsign-foundation/at_server/blob/trunk/at_secondary/at_secondary_server/pubspec.yaml) that brought me here, the problem isn't with dependencies on pub.dev but only happens with git references.

So these are fine:

```yaml
dependencies:
  args: ^1.5.0
  uuid: ^3.0.4
  convert: ^2.1.1
  crypto: ^3.0.0
  crypton: ^2.0.1
  collection: ^1.14.13
  basic_utils: ^3.0.0
```

But this causes the issue:

```yaml
  at_persistence_secondary_server:
    git:
      url: https://github.com/atsign-foundation/at_server.git
      path: at_secondary/at_persistence_secondary_server
      ref: trunk
      version: ^1.0.1+7
```

So the blast radius for this being a problem is diminished, but still likely to be meaningful.

**Comment by Derrick56007** [comment]
experiencing the same issue when using a git dependency

**Comment by subfuzion** [comment]
Adding git to the base image definitely makes sense for handling git dependencies; adding it is certainly a better experience than asking the community to create their own images to address this use case.

**Comment by athomas** [comment]
FYI, this will get published to docker hub with the upcoming 2.13.3 release later this week.

**Comment by Dokotela** [comment]
I have a follow-up question to this. I have a git dependency in my dart app, but it's to a private repo. I've been trying to put credentials in my Docker file, but haven't been successful. Has anyone been able to do this?
